postgres-driver/postgres_driver.go

```diff
@@ -127,12 +174,95 @@ func (d *PostgresDriver) WriteTransactions(txs []*indexer.Transaction) error {
 	return nil
 }
 
-// ReadTransactions returns all transactions on the database
-// TODO: add pagination
-func (d *PostgresDriver) ReadTransactions() ([]*indexer.Transaction, error) {
+// ReadTransactionsOptions optional parameters for ReadTransactions
+type ReadTransactionsOptions struct {
+	PerPage int
+	Page    int
+}
+
+// ReadTransactions returns transactions on the database with pagination
+// Optinal values defaults: page: 1, perPage: 1000
+func (d *PostgresDriver) ReadTransactions(options *ReadTransactionsOptions) ([]*indexer.Transaction, error) {
+	perPage := defaultPerPage
+	page := defaultPage
+
+	if options != nil {
+		perPage = getPerPageValue(options.PerPage)
+		page = getPageValue(options.Page)
+	}
+
+	move := getMoveValue(perPage, page)
+
+	tx, err := d.Beginx()
+	if err != nil {
+		return nil, err
+	}
+
+	query := fmt.Sprintf(selectTransactionsScript, move, perPage)
+
+	var transactions []*dbTransaction
+
+	err = tx.Select(&transactions, query)
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	var indexerTransactions []*indexer.Transaction
+
+	for _, dbTransaction := range transactions {
+		indexerTransactions = append(indexerTransactions, dbTransaction.toIndexerTransaction())
+	}
+
+	return indexerTransactions, nil
+}
+
+// ReadTransactionsByFromAddressOptions optional parameters for ReadTransactionsByFromAddress
+type ReadTransactionsByFromAddressOptions struct {
+	PerPage int
+	Page    int
+}
+
+func validateAddress(fromAddress string) bool {
+	return addressRegex.MatchString(fromAddress) && len(fromAddress) == addressLength
```

Comment:
this sounds like an util function for pocket-go

Reply:
U right